pkg/controller.v2/trainjob_controller.go

```diff
@@ -82,37 +81,20 @@ func (r *TrainJobReconciler) createOrUpdateObjs(ctx context.Context, trainJob *k
 	if err != nil {
 		return err
 	}
+
 	for _, obj := range objs {
-		var gvk schema.GroupVersionKind
-		if gvk, err = apiutil.GVKForObject(obj.DeepCopyObject(), r.client.Scheme()); err != nil {
-			return err
+		// TODO: evaluate the fields we need in apply configuration from the object.
+		//applyConfig := v2alpha1applyconfiguration.TrainJob(trainJob.Name, trainJob.Namespace).WithSpec(
+		//	v2alpha1applyconfiguration.TrainJobSpec())
+
+		if err := r.client.Patch(ctx, obj, client.Apply, client.FieldOwner("trainjob-controller"), client.ForceOwnership); err != nil {
```

Comment:
@tenzen-y I'm wondering if we need to support applyconfigurations right away when we aren't sure of the exact fields that need to be specified? Even if we need to use applyconfiguration instead of passing the object, controller-runtime doesn't support it natively, we will have to pass on kubeclient instead. 

`r.client.Patch` should take care of the create+update for now (I'll look into filtering out [metadata fields](https://github.com/kubernetes-sigs/cluster-api/blob/578b70f79659003a005f390cc022cf17f151cebc/internal/util/ssa/patch.go#L74) though)

Reply:
> I'm wondering if we need to support applyconfigurations right away when we aren't sure of the exact fields that need to be specified?

I agree with you. I do not imagine introducing the applyconfiguration now. Let's just directly use client.Object.

> r.client.Patch should take care of the create+update for now (I'll look into filtering out [metadata fields](https://github.com/kubernetes-sigs/cluster-api/blob/578b70f79659003a005f390cc022cf17f151cebc/internal/util/ssa/patch.go#L74) though)

I think so too. 
I'm wondering if we can just copy objectMeta (only for uid, name, namespace, generation), typeMeta, and GVK like this: https://github.com/kubernetes-sigs/kueue/blob/90ef60760b849e475f7cf32d07669bb91bbb479f/pkg/workload/workload.go#L503

WDYT?

Reply:
Makes sense. I'll rebase this PR.